### PR TITLE
Fix cut / delete for content with lock created by current user

### PR DESCRIPTION
**Pull request by laulaz**
This refs #266 & https://github.com/plone/Products.CMFPlone/issues/3846

### CODE REVIEW

**Comment by laulaz** [comment]
@jenkins-plone-org please run jobs

File: plone/app/content/browser/actions.py [hunk 1]
```diff
@@ -251,16 +260,24 @@ def do_redirect(self, url, message=None, message_type="info", raise_exception=No
...
-                        "title": self.title,
-                    },
-                ),
-            )
+        try:
+            lock_info = self.context.restrictedTraverse("@@plone_lock_info")
+        except AttributeError:
+            lock_info = None
+        if lock_info is not None:
+            if lock_info.is_locked_for_current_user():
```
Comment (jensens) [inline plone/app/content/browser/actions.py:268]: Isn't the logic here in the wrong direction/wrong?

**Review by jensens — CHANGES_REQUESTED** [review]

File: plone/app/content/browser/actions.py (hunk 1 above)
Comment (jensens) [reply plone/app/content/browser/actions.py:268]: Thanks for clarification, this is indeed confusing....

**Review by jensens — APPROVED** [review]

**Review by davisagli — APPROVED** [review]
It would be nice to have a solution in Zope's functions for cutting and deleting, so that it works regardless of what UI is being used and also if a sub-item is locked. But, I acknowledge that could be complicated and this looks like a reasonable improvement.

**Pull request merged**